### ISSUE TITLE
resolve window.require('path') dependencies

### ISSUE THREAD

**Issue by tvolodimir** [sverweij/dependency-cruiser#206]
Electron.js app has a specific approach for using at compile-time (webpack bundling, etc.) and runtime dependencies. So for example, webpack bundles sources and don't handle 'window.require' functions. But at runtime 'window.require' works like in node.js 'require'

### Context
```ts
//file ./src/index.ts
import { Component } from '@angular/core'; // <-- this is dependent module at compile-time 
...
export class AboutNgComponent {
  private electron: any;
  public get version(): string {
    return this.electron.remote.app.getVersion();
  }
  construnctor() {
    this.electron = window.require('electron');  // <-- this is dependent module at runtime
  }
}

```

### Expected Behavior
handle 'window.require' dependencies

### Current Behavior
doesn't find 'window.require' modules

### Possible Solution
I can't use 'require' instead of 'window.require' - see. https://github.com/electron/electron/issues/7300
So I hope, there are ability to add some extensions to handle specific cases like current

**Comment by sverweij** [comment]
Hi @tvolodimir it's indeed not yet possible to recognize `window.require` in addition to other types of imports, but it is doable to add support for it.

There's some things that need looking into:
- is the electron platform the only one having a 'special' import function? If so, we might need to make it configurable - maybe also to cover scenarios where require is renamed or wrapped.
- Can we do this blindly on all platforms without loss of performance and result-fidelity?
- If not - is it possible to auto-detect whether window.require (or any of other special import functions) needs to be taken into account?

**Comment by sverweij** [comment]
hi @tvolodimir dependency-cruiser@beta currently has the ability to recognise requires-that-are-not-requires, like the `window.require` in your electron source code. It's a more generic solution that also captures require wrappers:

- In the config you tell dc what exotic require strings to expect. In your case, add  `"exoticRequireStrings": ["window.require"]` to the options in your dependency-cruiser config and Bob's your uncle. See [exoticRequireStrings](https://github.com/sverweij/dependency-cruiser/blob/develop/doc/rules-reference.md#exotic-ways-to-require-modules---exoticrequirestrings) for details.

- It's also possible to use the fact an 'exotic require' (like window.require) was used to import a module in rule definitions - see [exoticRequire and exoticRequireNot](https://github.com/sverweij/dependency-cruiser/blob/develop/doc/rules-reference.md#exoticrequire-and-exoticrequirenot)

**Comment by tvolodimir** [comment]
@sverweij great work 😀🎉💪, Its perfectly fits my current project with electron, angular, monorepa, symlinks, pnpm, typescript, 

I checked and it works well with
```
"forbidden": [
  {
    "name": "ban-all-exotic-requires",
    "severity": "error",
    "from": {},
    "to": {
      "exoticRequire": ".+"
    }
  }
],
"options": {
  "exoticRequireStrings": ["window.require"]
}
```

**Comment by sverweij** [comment]
(just published in the main branch as version 5.4.0 - see [release notes](https://github.com/sverweij/dependency-cruiser/releases/tag/v5.4.0) for details)
